nixos/modules/services/backup/restic.nix

```diff
@@ -296,9 +314,21 @@ in
             excludeFlags = optional (backup.exclude != []) "--exclude-file=${pkgs.writeText "exclude-patterns" (concatStringsSep "\n" backup.exclude)}";
             filesFromTmpFile = "/run/restic-backups-${name}/includes";
             doBackup = (backup.dynamicFilesFrom != null) || (backup.paths != null && backup.paths != []);
+            niceness =
+              let
+                ioniceClass = {
+                  "idle" = 3;
+                  "best-effort" = 2;
+                  "realtime" = 1;
+                }.${backup.ioSchedulingClass};
+              in
+                ''${pkgs.util-linux}/bin/ionice -t ''
+                + '' -c ${toString ioniceClass} ''
+                + '' ${optionalString (backup.ioPriority != null) "-n ${toString backup.ioPriority}"} ''
+                + '' ${optionalString (backup.niceness >= 1) "${pkgs.coreutils}/bin/nice -n ${toString backup.niceness}"} '';
```

Comment:
Regarding `backup.niceness >= 1`: Why only allow a positive niceness, when even setting the right restrictions within the option's type. No, I would also expect a negative niceness to work, i.e., when the backup's CPU priority is low.